### PR TITLE
Pass --file to `conda config` to avoid writing to `~/.condarc`

### LINKED ISSUE #140: Conda.jl probably shouldn't modify the `~/.condarc` file
It looks like adding channels to the Conda.jl rootenv will modify the users `.condarc` file. Conda.jl should probably just maintain its own file.

### PR DESCRIPTION
**Pull request by rofinn**
Appears to fix #140

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -46,6 +46,11 @@ Conda.add_channel("foo", env)
 # Testing that calling the function twice do not fail
 Conda.add_channel("foo", env)
 
+@test occursin("foo", read(Conda.conda_rc(env), String))
+homerc = joinpath(homedir(), ".condarc")
+
+@test !isfile(homerc) || !occursin("foo", read(homerc, String))
+
 Conda.rm_channel("foo", env)
```
[the remark this answers is not in the document]
Comment (omus) [reply test/runtests.jl]: Maybe something like:
```julia
condarc_path = joinpath(homedir(), ".condarc")
last_modified = stat(condarc_path).mtime

Conda.add_channel("foo", env)
@test Conda.channels(env) == ["foo", "defaults"]

# Testing that calling the function twice do not fail
Conda.add_channel("foo", env)

# Validate that only the Conda.jl RC file was modified
@test occursin("foo", read(Conda.conda_rc(env), String))
@test !isfile(condarc_path) || !occursin("foo", read(condarc_path, String))

Conda.rm_channel("foo", env)
@test stat(condarc_path).mtime == last_modified
```

**Comment by rofinn** [comment]
I was having difficulting tracking down all the changes between conda 4.5 and 4.6, but it seems like the env is still useful for finding the file, but something in 4.6 is broken where it always defaults to writing the the `~/.configrc` file now unless you overwrite it with `--file`.

**Comment by tkf** [comment]
I see. It'd be nice to report upstream if it hasn't reported yet.

**Comment by rofinn** [comment]
I didn't seen an open issue for that specifically, but I can make one. Since I don't see any reference to these variables in the documentation it's possible that they're simply dropping a non-public feature?

**Comment by tkf** [comment]
Ah, that's possible.  I didn't realize that it's not documented.  If that's the case I think we should automatically add `--file ...` option in `_set_conda_env` instead of doing that in each call site.

**Comment by rofinn** [comment]
I think it's only an option for `conda config` though.

**Pull request merged**